CRM/Activity/BAO/Activity.php

```diff
@@ -317,6 +317,8 @@ public static function create(&$params) {
 
     // CRM-9137
     if (!empty($params['id'])) {
+      $activity = CRM_Activity_DAO_Activity::findById($params['id']);
+
       CRM_Utils_Hook::pre('edit', 'Activity', $activity->id, $params);
```

Comment:
I don't believe all the necessary params are copied into the object - it only copies the newly edited/submitted params.

For example, if a user inline edits an Activity subject, the Activity object that gets passed to the post hook does not contain the `activity_type_id` property and they are unable to switch on this value.